pkg/provider/acme/local_store.go

```diff
@@ -111,6 +120,15 @@ func (s *LocalStore) listenSaveAction() {
 	})
 }
 
+// unSafeCopyOfStoredData creates maps copy of source. Is not thread safe, you should use `s.lock`.
+func (s *LocalStore) unSafeCopyOfStoredData(source map[string]*StoredData) map[string]*StoredData {
+	result := map[string]*StoredData{}
+	for k, v := range source {
+		result[k] = v
+	}
+	return result
+}
+
```

Comment:
Re Hello @walkline,

You are right about the deadlock.
Still, i would use the receiver since i do not see side effect to that, and it's more consistent for a method.

```suggestion
// unSafeCopyOfStoredData creates maps copy of source. Is not thread safe, you should use `s.lock`.
func (s *LocalStore) unSafeCopyOfStoredData() map[string]*StoredData {
	result := map[string]*StoredData{}
	for k, v := range s.storedData {
		result[k] = v
	}
	return result
}
```